packages/lesswrong/components/tagging/TagEditsTimeBlock.tsx

```diff
@@ -48,21 +50,37 @@ const TagEditsTimeBlock = ({before, after, reportEmpty, classes}: {
       reportEmpty();
     }
   }, [loading, data, reportEmpty]);
+  const [expanded, setExpanded] = useState(false)
+  
+  let tagUpdatesInTimeBlock = [...(data?.TagUpdatesInTimeBlock || [])]
```

Comment:
What does all this do?

Reply:
Makes a copy of the array to avoid the fact that apollo hands us a read-only version. `[...arr]` is just a hacky way to do that using [array spreading](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). The `|| []` is because spreading is not safe for null values\[1], so we need to force it to be an array

\[1]: See

```js
> foo = null
null
> [...foo]
Uncaught TypeError: foo is not iterable
```

Reply:
Object spreading is nullsafe — javascript is not a good language. Thanks to @jimrandomh for complaining about this fact a long time ago which helped me remember it here.